### PR TITLE
add Prompt Event logging to dashboard

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: app/models/prompt_event.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  RESPONSE_VALUES = %w( Yes No )
+  TYPES = { prompt: 'Prompt', response: 'Response' }
+
+  belongs_to :intervention_module
+  belongs_to :participant
+
+  validates :intervention_module, :participant, :occurred_at, presence: true
+  validates :response_value, inclusion: { in: RESPONSE_VALUES, allow_nil: true }
+
+  delegate :title, to: :intervention_module, prefix: true
```
Comment (eschlange) [inline app/models/prompt_event.rb]: I haven't used the prefix option for 'delegate'. Does this prepend 'intervention_module' onto the delegated method name?

Comment (ericcf) [inline app/models/prompt_event.rb]: Yes, it creates the method `intervention_module_title`: http://api.rubyonrails.org/classes/Module.html#method-i-delegate

File: app/models/prompt_event_reading.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# A prompt event from Purple Robot displays a prompt to a Participant.
+class PromptEventReading
+  def self.save(reading)
+    value = reading.attributes['FEATURE_VALUE']
+    i_module = InterventionModule.find_by_title(value['intervention'])
+    token = TokenAuth::AuthenticationToken
+            .find_by(client_uuid: value['client_uuid'],
+                     value: value['authentication_token'])
+
+    return false unless token
```
Comment (eschlange) [inline app/models/prompt_event_reading.rb]: This is probably just my dislike for multiple exit points for a method, but this is confusing to me. I think just a simple conditional here would make more sense, something like:
```
if token
  PromptEvent.create(
    participant_id: token.entity_id,
    intervention_module: i_module,
    response_value: value['response_value'],
    occurred_at: reading.timestamp
  )
else
  false
end
```

**Comment by eschlange** [comment]
Everything else looks good!

File: app/models/prompt_event_reading.rb (hunk 2 above)
Comment (ericcf) [inline app/models/prompt_event_reading.rb]: The code as is has one explicit and one implicit `return`, whereas your example has two implicit `returns`. To get it down to one return, maybe something like this?

```ruby
!token.nil? && PromptEvent.create(...
```

File: app/controllers/participants_controller.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # View Participants.
 class ParticipantsController < ApplicationController
-  DAYS_OF_RESPONSES = 14
+  DURATION_OF_RESPONSES = 14.days
```
Comment (michaelwehrley) [inline app/controllers/participants_controller.rb]: I have a question.  calling `days` on 14 doesn't feel like a constant.  I am not sure what it returns without testing it (maybe that doesn't matter) and what if return value of the function changes?  I like what you originally had.  Why did you change it?

Comment (ericcf) [inline app/controllers/participants_controller.rb]: Everything in Ruby is an object, so `14` is an instance of `Fixnum`, and `14.days` is an instance of `ActiveSupport::Duration`. I changed it here because I found myself calling `DURATION_OF_RESPONSES.days` in a few places which was redundant, and it also seemed more meaningful to make the constant self-contained so that there was no ambiguity as to what the units were.
